consensus/src/sync/sync_queue.rs

```diff
@@ -452,14 +450,24 @@ where
                         }
                     }
                 }
-                None => {
+                Poll::Ready(None) => {
                     return if self.ids_to_request.is_empty() || self.peers.read().is_empty() {
-                        Poll::Ready(None)
+                        self.waker.store_waker(cx);
+                        Poll::Pending
```

Comment:
The return could be moved out of the conditional as they are now the same. As half the condition is repeated in the else case this can maybe also be combined.

Reply:
Did the former. Not sure how to do the latter.